src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1388,6 +1390,248 @@ public async Task TestSchemaIntrospectionQuery(bool enableIntrospection, bool ex
+            request.Headers.Add("accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9");
+
+            HttpResponseMessage response = await client.SendAsync(request);
+            Assert.AreEqual(expectedStatusCode, response.StatusCode);
+            string actualBody = await response.Content.ReadAsStringAsync();
+            Assert.IsTrue(actualBody.Contains(expectedContent));
+        }
+
+        /// <summary>
+        /// Validates the OpenAPI documentor behavior when enabling and disabling the global REST endpoint
+        /// for the DAB engine.
+        /// Global REST enabled:
+        /// - POST to /openapi triggers OpenAPI document creation but should fail with 405 Method Not Allowed.
+        /// - GET to /openapi returns the created OpenAPI document, but should fail with 404 Not Found.
+        /// Global REST disabled:
+        /// - POST to /openapi triggers OpenAPI document creation fails with 409 Conflict because the
+        /// document was already created during engine startup.
+        /// - GET to /openapi returns the created OpenAPI document and succeeds with 200 OK.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(true, false, DisplayName = "Global REST endpoint enabled - conflict creating OpenAPI doc and successful retrieval")]
+        [DataRow(false, true, DisplayName = "Global REST endpoint disabled - failure to create and retrieve OpenAPI doc")]
+        [TestCategory(TestCategory.MSSQL)]
+        public async Task OpenApi_GlobalEntityRestPath(bool globalRestEnabled, bool expectsError)
+        {
+            Dictionary<GlobalSettingsType, object> settings = new()
+            {
+                { GlobalSettingsType.GraphQL, JsonSerializer.SerializeToElement(new GraphQLGlobalSettings(){ Enabled = true }) },
+                { GlobalSettingsType.Rest, JsonSerializer.SerializeToElement(new RestGlobalSettings(){ Enabled = globalRestEnabled }) }
+            };
+
+            DataSource dataSource = new(DatabaseType.mssql)
+            {
+                ConnectionString = GetConnectionStringFromEnvironmentConfig(environment: TestCategory.MSSQL)
+            };
+
+            RuntimeConfig configuration = InitMinimalRuntimeConfig(globalSettings: settings, dataSource: dataSource);
+            const string CUSTOM_CONFIG = "custom-config.json";
+            File.WriteAllText(
+                CUSTOM_CONFIG,
+                JsonSerializer.Serialize(configuration, RuntimeConfig.SerializerOptions));
+
+            string[] args = new[]
+            {
+                    $"--ConfigFileName={CUSTOM_CONFIG}"
+            };
+
+            using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+            using (HttpClient client = server.CreateClient())
+            {
+                // Setup and send request
```

Comment:
similar comment on line 1492 and 1518, but different operation. Maybe mention `Sending a GET Request` and `Sending a POST request`

Reply:
Corrected (removed POST request as no longer needed because we generate doc during startup or during late config hydration, not user initiated.).

I added clarification comment that it is a GET request.